drivers/cc1200/cc1200.c

```diff
@@ -0,0 +1,361 @@
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_cc1200
+ * @{
+ * @file
+ * @brief       Basic functionality of cc1200 driver
+ *
+ * @author    Anon Mall <anon.mall@gt-arc.com>
+ * @}
+ */
+
+#include "luid.h"
+#include "board.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "xtimer.h"
+#include "cpu.h"
+#include "log.h"
+
+#include "cc1200.h"
+#include "cc1200-defaultsettings.h"
+#include "cc1200-defines.h"
+#include "cc1200-interface.h"
+#include "cc1200-internal.h"
+#include "cc1200-spi.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* Internal function prototypes */
+#ifndef CC1200_DONT_RESET
+static void _reset(cc1200_t *dev);
+static void _power_up_reset(cc1200_t *dev);
+#endif
+
+int cc1200_setup(cc1200_t *dev, const cc1200_params_t *params)
+{
+    DEBUG("%s:%s:%u\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+
+#ifdef MODULE_CC1200_HOOKS
+    cc1200_hooks_init();
+#endif
+
+
+    dev->params = *params;
+
+    /* Configure chip-select */
+    spi_init(dev->params.spi);
+    int spi_return = spi_init_cs(dev->params.spi, dev->params.cs);
+    if(spi_return != SPI_OK){
+        DEBUG("%s:%s:%u spi not ok\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
+    }
+
+
+
+#ifndef CC1200_DONT_RESET
+    /* reset device*/
+    _power_up_reset(dev);
+#endif
+
+    /* set default state */
+    dev->radio_state = RADIO_IDLE;
+
+    /* Write configuration to configuration registers */
+    cc1200_writeburst_reg(dev, 0x00, cc1200_default_conf, cc1200_default_conf_size);
+
+    /* Write extended config for CC1200 */
+    cc1200_write_reg(dev, CC1200_IF_MIX_CFG, 0x18);
+    cc1200_write_reg(dev, CC1200_TOC_CFG, 0x03);
+    cc1200_write_reg(dev, CC1200_MDMCFG2, 0x02);
+    cc1200_write_reg(dev, CC1200_FREQ2, 0x56);
+    cc1200_write_reg(dev, CC1200_FREQ1, 0xCC);
+    cc1200_write_reg(dev, CC1200_FREQ0, 0xCC);
+    cc1200_write_reg(dev, CC1200_IF_ADC1, 0xEE);
+    cc1200_write_reg(dev, CC1200_IF_ADC0, 0x10);
+    cc1200_write_reg(dev, CC1200_FS_DIG1, 0x04);
+    cc1200_write_reg(dev, CC1200_FS_DIG0, 0x50);
+    cc1200_write_reg(dev, CC1200_FS_CAL1, 0x40);
+    cc1200_write_reg(dev, CC1200_FS_CAL0, 0x0E);
+    cc1200_write_reg(dev, CC1200_FS_DIVTWO, 0x03);
+    cc1200_write_reg(dev, CC1200_FS_DSM0, 0x33);
+    cc1200_write_reg(dev, CC1200_FS_DVC1, 0xF7);
+    cc1200_write_reg(dev, CC1200_FS_DVC0, 0x0F);
+    cc1200_write_reg(dev, CC1200_FS_PFD, 0x00);
+    cc1200_write_reg(dev, CC1200_FS_PRE, 0x6E);
+    cc1200_write_reg(dev, CC1200_FS_REG_DIV_CML, 0x1C);
+    cc1200_write_reg(dev, CC1200_FS_SPARE, 0xAC);
+    cc1200_write_reg(dev, CC1200_FS_VCO0, 0xB5);
+    cc1200_write_reg(dev, CC1200_IFAMP, 0x05);
+    cc1200_write_reg(dev, CC1200_XOSC5, 0x0E);
+    cc1200_write_reg(dev, CC1200_XOSC1, 0x03);
+    cc1200_write_reg(dev, CC1200_AGC_CS_THR, (uint8_t) -91);
```

Comment:
Please define these default values somewhere, as you already did for the channel.